lib/td/client/api/job.rb

```diff
@@ -287,6 +248,114 @@ def query(q, type=:hive, db=nil, result_url=nil, priority=nil, retry_limit=nil,
 
   private
 
+  def validate_content_length(response, current_total_chunk_size)
+    if expected_size = response.header['Content-Range'][0]
+      expected_size = expected_size[/\d+$/].to_i
+    elsif expected_size = response.header['Content-Length'][0]
+      expected_size = expected_size.to_i
+    end
+
+    if expected_size.nil?
+    elsif current_total_chunk_size < expected_size
+      # too small
+      raise EOFError, 'httpclient IncompleteError'
+    elsif current_total_chunk_size < expected_size
+      # too large
+      raise_error("Get job result failed", res)
```

Comment:
`job_result_download` checks status before calling `validate_content_length_with_range`.

As you say `too large` is odd situation, which is caused by a bug of client or server.
On such case, I want to have HTTP information.